src/main/java/picard/illumina/ExtractIlluminaBarcodes.java

```diff
@@ -860,16 +860,26 @@ private boolean ensureLookupMinimumValue(final byte[][] qualityScores, final int
         /**
          * Find the best barcode match for the given read sequence, and accumulate metrics
          *
+         * NOTE: the returned BarcodeMatch object will contain mismatches mismatchesToSecondBest values that may be
+         * inaccurate as long as the conclusion match/no-match isn't affected. for example, mismatches and mismatchesToSecondBest
+         * may be smaller than their true value if mismatches is truly larger than maxMismatches.
+         * Also, mismatchesToSecondBest might be smaller than it's true value if its true value is greater than
+         * mismatches + minMismatchDelta. This is due to an optimization which allows the distance calculation to stop once
+         * the conclusion (Match or no-Match) can be reached.
+         *
          * @param readSubsequences portion of read containing barcode
          * @return perfect barcode string, if there was a match within tolerance, or null if not.
+         *
          */
-        private BarcodeMatch findBestBarcode(final byte[][] readSubsequences,
+        static BarcodeMatch findBestBarcode(final byte[][] readSubsequences,
                                              final byte[][] qualityScores,
                                              final Map<String, BarcodeMetric> metrics,
                                              final int maxNoCalls,
                                              final int maxMismatches,
                                              final int minMismatchDelta,
-                                             final int minimumBaseQuality) {
+                                             final int minimumBaseQuality,
+                                            Map<String, BarcodeMatch> barcodeLookupMap,
```

Comment:
Might be better to move the `barcodeLookupMap` checking and updating into a separate private non-static method which can then call this static method if it can't find the barcode in the lookup table, instead of having to pass a lookup table to this method.